sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -294,6 +327,9 @@ def message_to_proto_str(element):
       raise TypeError(
           'Unexpected element. Type: %s (expected: PubsubMessage), '
           'value: %r' % (type(element), element))
+    # message_id and publish_time cannot be written back to pubsub
```

Comment:
can you tell me more about this? will pubsub reject the message? would it make sense to make a copy of the message to avoid mutating the original message?

Reply:
My understanding is that yes, pubsub will reject the message.  https://cloud.google.com/pubsub/docs/reference/rpc/google.pubsub.v1#pubsubmessage says, in re both fields:

> It must not be populated by the publisher in a Publish call.

Fair question about copying vs mutating, and I maybe don't know enough about beam's internals to have a real answer? Logically WriteToPubsub is (I think) going to be the end of the line for the message since it doesn't return a pcollection, but is the message purely in local state here, or does mutating it potentially change it in-flight for other transforms that might be processing it via other routes at the same time?

Reply:
actually yeah, putting this into `_to_proto_str` makes a lot of sense and that way we don't eat memory copying the message. 👍